nautobot/dcim/forms.py

```diff
@@ -3235,8 +3232,17 @@ def __init__(self, *args, **kwargs):
             # Limit VLAN choices by Location
             if locations.count() == 1:
                 location = locations.first()
-                self.fields["untagged_vlan"].widget.add_query_param("locations", location.pk)
+                # In the case of a single location, use the available_on_device query param to limit untagged VLAN choices
+                # to those available on the devices in that location and in the ancestors of the location.
+                device_pks = [device.pk for device in devices]
+                self.fields["untagged_vlan"].widget.add_query_param("available_on_device", device_pks)
                 self.fields["tagged_vlans"].widget.add_query_param("locations", location.pk)
+            else:
+                self.fields["untagged_vlan"].widget.add_query_param("locations", "null")
+                if locations.count() > 1:
```

Comment:
Isn't this implied by the above `else`? Or is this just specifically covering the case where `locations` might be empty?

Reply:
Yeah think about it more, this might not be necessary after all. I wrote it specifically for the case we might have multiple interfaces from multiple locations. But I think the location.count() can only be 0 or 1.